python/cudf/cudf/core/index.py

```diff
@@ -2811,57 +2813,80 @@ def interval_range(
             "Of the four parameters: start, end, periods, and "
             "freq, exactly three must be specified"
         )
-    if not isinstance(start or freq or end, int) and not isinstance(
-        start or freq or end, float
+    args = [start, end, freq, periods]
```

Comment:
Should use a tuple instead of a list since we're not mutating it
```suggestion
    args = (start, end, freq, periods)
```